### PR TITLE
Add test for retention and fix bugs about it

### PR DESCRIPTION
**Pull request by ddurieux**

### CODE REVIEW

File: alignak/scheduler.py [hunk 1]
```diff
@@ -1329,126 +1360,150 @@ def get_retention_data(self):
...
+    def restore_retention_data(self, data):
         """Restore retention data
 
         Data coming from retention will override data coming from configuration
         It is kinda confusing when you modify an attribute (external command) and it get saved
         by retention
 
-        :param data:
-        :type data:
+        :param data: data fron retention
```
Comment (mohierf) [inline alignak/scheduler.py]: typo

File: alignak/scheduler.py [hunk 2]
```diff
@@ -1329,126 +1360,150 @@ def get_retention_data(self):
...
+            else:
+                comment_type = 2
+            data = {
+                'persistent': item.acknowledgement.persistent,
+                'author': item.acknowledgement.author,
+                'comment': item.acknowledgement.comment, 'comment_type': comment_type,
+                'entry_type': 4, 'source': 0, 'expires': False, 'expire_time': 0, 'ref': item.uuid
+            }
+        # Relink the notified_contacts as a set() of true contacts objects
+        # it it was load from the retention, it's now a list of contacts
```
Comment (mohierf) [inline alignak/scheduler.py]: typo: if it was loaded from ...

File: test/test_retention.py [hunk 3]
```diff
@@ -58,49 +59,100 @@ def test_scheduler_get_retention(self):
         self.scheduler_loop(1, [[host, 2, 'DOWN'], [svc, 2, 'CRITICAL']])
         time.sleep(0.1)
 
+        now = time.time()
+        # downtime host
+        excmd = '[%d] SCHEDULE_HOST_DOWNTIME;test_host_0;%s;%s;1;0;1200;test_contact;My downtime' \
+                % (now, now + 120, now + 1200)
+        self.schedulers['scheduler-master'].sched.run_external_command(excmd)
+        self.external_command_loop()
+
```
Comment (mohierf) [inline test/test_retention.py]: You should assert to be sure the downtime is attached to the host, no?

Comment (ddurieux) [inline test/test_retention.py]: We have other test test if downtime is right attached to the item(host/service)

Comment (mohierf) [inline test/test_retention.py]: Ok

**Pull request merged**